vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/contextmenu/GridContextMenu.java

```diff
@@ -257,10 +282,14 @@ public void setDynamicContentHandler(
     protected boolean onBeforeOpenMenu(JsonObject eventDetail) {
         Grid<T> grid = (Grid<T>) getTarget();
         String key = eventDetail.getString("key");
+        String columnId = eventDetail.getString("columnId");
```

Comment:
Using the column ID is consistent with what you get from the opened change event, but requires an ID to be set on the column. Ideally we would always provide a column. I can imagine developers not familiar with the opened changed event getting confused why the column is null, and then having to dig around how to make it show up.

As the method for accessing the column by internal ID has protected visibility and is not accessible from this class, maybe we can introduce a construct to expose some internals from the Grid to be used by related components. Taking inspiration from `UI.getInternals`, we could add something like this to the Grid:
```java
    /**
     * Helper class for exposing internal Grid functionality to related components, such as {@link GridContextMenu}
     * <p>
     * For internal use only. May be renamed or removed in a future release.
     * @param <T> the item type of the grid
     */
    public static class GridInternals<T> {
        private final Grid<T> grid;

        public GridInternals(Grid<T> grid) {
            this.grid = grid;
        }

        public Column<T> getColumnByInternalId(String id) {
            return grid.getColumnByInternalId(id);
        }
    }

    /**
     * Exposes internals of the Grid to be used by other Grid-related Vaadin components.
     * <p>
     * For internal use only. May be renamed or removed in a future release.
     * @return helpers for accessing internals
     */
    public GridInternals<T> getInternals() {
        return new GridInternals<>(this);
    }
```

We could also re-use this to provide the column instance in the opened change event, and deprecate the `columnId` property.